### PR TITLE
🎁 Adding indices to Bulkrax Tables

### PR DESCRIPTION
**Pull request by jeremyf**
Note: this is a best guess of what could use indexing.  The guess is based on my recent querying.

- `add_index :bulkrax_entries, :identifier` :: because the identifier is an important concept for an import.
- `add_index :bulkrax_entries, :type` :: because filtering on the type of an entry is important.
- `add_index :bulkrax_entries, [:importerexporter_id, :importerexporter_type], name: 'bulkrax_entries_importerexporter_idx` :: this is a foreign key for the entries relationship to the importer/exporter.

- `add_index :bulkrax_pending_relationships, :parent_id` :: a foreign key
- `add_index :bulkrax_pending_relationships, :child_id`  :: a foreign key

- `add_index :bulkrax_statuses, [:statusable_id, :statusable_type], name: 'bulkrax_statuses_statusable_idx'` :: a foreign key
- `add_index :bulkrax_statuses, [:runnable_id, :runnable_type], name: 'bulkrax_statuses_runnable_idx'` :: a foreign key
- `add_index :bulkrax_statuses, :error_class` :: when querying what's in error, this index is helpful; also one used for grouping.

### CODE REVIEW

File: spec/test_app/config/database.yml [hunk 1]
```diff
@@ -4,22 +4,23 @@
 #   Ensure the SQLite 3 gem is defined in your Gemfile
 #   gem 'sqlite3'
 #
-default: &default
```
Comment (alishaevn) [inline spec/test_app/config/database.yml:7]: was this not working? or do you just have a preference of explicitly stating the variables for each environment?

File: db/migrate/20230608153601_add_indices_to_bulkrax.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    add_index :bulkrax_entries, :identifier
+    add_index :bulkrax_entries, :type
+    add_index :bulkrax_entries, [:importerexporter_id, :importerexporter_type], name: 'bulkrax_entries_importerexporter_idx'
+
+    add_index :bulkrax_pending_relationships, :parent_id
+    add_index :bulkrax_pending_relationships, :child_id
+
+    add_index :bulkrax_statuses, [:statusable_id, :statusable_type], name: 'bulkrax_statuses_statusable_idx'
+    add_index :bulkrax_statuses, [:runnable_id, :runnable_type], name: 'bulkrax_statuses_runnable_idx'
```
Comment (bkiahstroud) [inline db/migrate/20230608153601_add_indices_to_bulkrax.rb:11]: Is `_idx` a naming convention? If so, what does it signify?

Comment (jeremyf) [reply db/migrate/20230608153601_add_indices_to_bulkrax.rb:11]: For indices we want unit names; I add `_idx` as a short hand indicator that this database structure is an index.

File: spec/test_app/config/database.yml (hunk 1 above)
Comment (jeremyf) [reply spec/test_app/config/database.yml:7]: This was not working because of some Psych stuff.

**Review by bkiahstroud — APPROVED** [review]

**Comment by ShanaLMoore** [comment]
consider removing coverage task

**Pull request merged**